includes/transformer/class-activity-object.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Activity Object Transformer Class.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Transformer;
+
+/**
+ * Activity Object Transformer Class.
+ */
+class Activity_Object extends Base {
+	/**
+	 * Transform the WordPress Object into an ActivityPub Object.
+	 *
+	 * @return Base_Object The ActivityPub Object.
+	 */
+	public function to_object() {
+		return $this->transform_object_properties( $this->item );
+	}
+
+	/**
+	 * Get the ID of the WordPress Object.
+	 *
+	 * @return string The ID of the WordPress Object.
+	 */
+	protected function get_id() {
+		return '';
+	}
+
+	/**
+	 * Helper function to get the @-Mentions from the post content.
+	 *
+	 * @return array The list of @-Mentions.
+	 */
+	protected function get_mentions() {
+		/**
+		 * Filter the mentions in the post content.
+		 *
+		 * @param array   $mentions The mentions.
+		 * @param string  $content  The post content.
+		 * @param WP_Post $post     The post object.
+		 *
+		 * @return array The filtered mentions.
+		 */
+		return apply_filters(
+			'activitypub_extract_mentions',
+			array(),
+			$this->item->get_content() . ' ' . $this->item->get_summary(),
+			$this->item
+		);
+	}
+
+	/**
+	 * Returns a list of Mentions, used in the Post.
+	 *
+	 * @see https://docs.joinmastodon.org/spec/activitypub/#Mention
+	 *
+	 * @return array The list of Mentions.
+	 */
+	protected function get_cc() {
+		$cc       = array();
+		$mentions = $this->get_mentions();
+
+		if ( $mentions ) {
+			foreach ( $mentions as $url ) {
+				$cc[] = $url;
+			}
+		}
+
+		return $cc;
```

Comment:
```suggestion
		return $this->get_mentions() ?: array();
```
Could this be simplified to ^?
If `get_mentions()` always returned an array, the fallback would not be needed.

Reply:
We can remove the `if` and directly iterate gut we can't simply return `get_mention()` because it has some more data, not only the list of URLs.

Reply:
And we could use an `array_filter` instead!

Reply:
Hm, maybe `return array_values( $this->get_mentions() )`? 

Currently the function doesn't remove anything from `$mentions`, it just puts everything in `$cc`.

Reply:
🤔

Reply:
You are right, array_values should do the trick! It thought it has some more informations, but it's simply the key is the WebFinger handle and the valid the URL!